### ISSUE TITLE
Provide a way to set sender name per email address

### ISSUE THREAD

**Issue by x80486** [tutao/tutanota#516]
As a user i want to be able to configure a sender name per email alias so i don't expose my real name for any of my anonymous email addresses.

### Acceptance criteria

- Allow to set a different name for every email alias and the main email address.
- If there is no sender name configured for the used address, the sender name in the email should remain empty (no fallback).
- Sender name is used when writing an email (new email, reply, forward)
- Sender name is used in calendar invites/replies
- Sender name is used in external notification emails
- Sender name is stored encrypted with group key of the mailbox
- Sender name can be modified by the owner (even if non-admin) of the mailbox that the email address is assigned to.
- When creating a user the user name is also used as the sender name of the user's default email address
- When an admin creates an alias for a user (or for themselves), the sender name for the alias can also be specified. The user name is shown in the text field by default.
- When changing the user name or an email address name other names are not changed automatically (user name and email address names are independent)
- Existing sender names are retained (main sender name and user group). All email addresses inherit the configured sender name.
- Users can still log in with aliases

### Implementation hints

- Introduce configuration object per email address (users mail address and email aliases) which holds the name in the first place and later can also have the signature and out of office notification.
- Run a migration on the client that creates the config object and writes existing email addresses with name to it.
- Currently, a non-admin user can't see their aliases or change their sender name. We need to show alias list to all users and not only admins so that they can change their sender name.

GUI: Replace email alias table with new email address table/list were also the users main email address is listed. Show it also for non admin users. Allow specifying name but not activation/deactivation/deletion. One entry in the list should have two rows. First row shows email address, second row status and sender name were status is either "main address", "activated" or "deactivated"

## Tasks
 - [x] Design and change model
 - [x] Add client migration
 - [x] Add configuration UI to settings to configure the name per alias
 - [x] Allow non-admins to see mail aliases and configure the names for it
 - [x] Change the client to use the new model when sending mails/saving drafts
 - [x] Make sure that calendar invites include the right name
 - [x] Clarify with support/marketing that it behaves like expected
 - [x] Change UI for adding alias to allow configuring the name

## Test notes
 - [ ] Test new model: release new tutadb without schema upgrade and without a new client, test that spam report preference can still be changed

Prepare an existing account with admin, local admin and at least two normal users
 - [ ] Previously unmigrated account, log in, open mail editor. Sender name dropdown should pick up existing sender name
 - [ ] Go to settings and change sender name for some mail addresses. It should now be used in mail editor
 - [ ] Check that it is used in calendar event invites
 - [ ] Check that it is used in external notification emails
 - [ ] Check that you can add an alias with sender name, it should default to user name
 - [ ] Check that you can still edit the user name and it doesn't change sender name
 - [ ] Log in as a global admin, check that your own sender names can be changed via mail or user management settings
 - [ ] Check that you can view and change previous user's sender names
 - [ ] Check another user (user B) (who hasn't logged in yet). Should be able to do the same things
 - [ ] Check that you can add an alias with sender name, it should default to user name
 - [ ] Log in as a local admin. Check that you can also view and edit things for both users.
 - [ ] Log in as user B, check that the settings by admins have been applied, you can view and edit sender names
 - [ ] Check that mail address table shows primary mail address and email aliases and their statuses correctly
 - [ ] Check that as non-admin you don't see how many aliases you can add nor you can add/remove aliases
 - [ ] Check that number of available aliases is correctly updated when an alias is added
 - [ ] Check that as an admin aliases can be enabled/disabled and the status is displayed correctly
 - [ ] Check that whitelabel aliases can be deleted but not disabled

**Comment by mpfau** [comment]
The user must be able to configure the names (not only the admin). The user name is not used as a name for the mail address anymore.

**Comment by x80486** [comment]
I think so. I only have one user, so it's the `admin`.

This one was towards the ability to set a different "sender name" for all the registered e-mail addresses (including aliases). If you go to `Settings > User Management`, there is only one `Sender name`; the value there will be used for all outgoing e-mails, regardless of the alias in use.

I think that has to be fixed because not all aliases will/should use the same `Sender name` value – the same applies for the main account address.

**Comment by itisafrickinhighlander** [comment]
This suggestion is similar to what is normally present in the UX of provider or an e-mail client (which can do this independently from the provider):

![image](https://user-images.githubusercontent.com/54138438/63581112-8e9e0e00-c596-11e9-9c95-841b87832cca.png)

A specific example (and compliant with the privacy mindset) would be this - one wants to create an identity which is not real (but sounds legit) for services where one does't feel comfortable sharing and/or is not required to provide a real name, but considers these services legitimate enough that one may want to still preserve some shape of form of an identity, especially when there is an e-mail correspondence (contacting support, sales,...).

In a nutshell, one can still do it, but only via aforementioned account-wide Sender name option in the settings, which I would be genuinely surprised if anyone had it filled in as it would "sign" every sent e-mail with that name, which we all could then use aliases with a standard firstname+surname syntax.

**Comment by jeffmcneill** [comment]
This is the only issue stopping me from using Tutanota. I've got five aliases, but they have different names. Not being able to have a unique name per alias makes that functionality much less... functional. Creating and paying for five different accounts and then going through the login/logout logistics is not a great workaround. Any idea on when improving this feature is going to be available?

**Comment by C666PO** [comment]
I just started using Tutanota and i'm not sure if i will continue. Protonmail allowed me to have a different sender name for each email as also filtering emails by email address. I think it is critical for tutanota to implement this feature. I really hope they do it soon. It's the only think i miss in protonmail.

**Comment by 0Ky** [comment]
[**@mpfau**](https://github.com/mpfau) commented on [Nov 14, 2018, 8:01 PM GMT+11](https://github.com/tutao/tutanota/issues/516#issuecomment-438587608 "2018-11-14T09:01:27Z - Replied by Github Reply Comments"):
> The user must be able to configure the names (not only the admin). The user name is not used as a name for the mail address anymore.

Yes, a user can only configure 1 _**sender name**_ per account (they can change the sender name to what ever they like).

The problem here is that all aliases use the same sender name, we want to assign different _**sender name**_ simultaneously to every _**email aliases**_ created by a single tutanota user.

It has been few years and this feature has not yet been implemented. This was proposed in 2015 or 2016 on _**[tutanota.uservoice\.com](https://tutanota.uservoice.com/forums/237921-general/suggestions/8810773-set-a-different-name-for-alias)**_ (website is <ins>not in service</ins>, there's an archived page with an idea title of _[Set a different «Name» for alias](https://web.archive.org/web/20160731191224/https://tutanota.uservoice.com/forums/237921-general/filters/top)_) with hundreds of likes and possibly thousands or more by now.

We have been informed that it was _planned_, but haven't heard anything since then...

**Comment by charlag** [comment]
@0Ky what also to say about it other than "it's planned, look at the [roadmap](https://tutanota.com/roadmap/)"?
Please take into account that some things from top of it are already implemented and are either in beta or about to be released

**Comment by tukiab** [comment]
+1

**Comment by nikolayhg** [comment]
+1! Feature in demand!

**Comment by szmt** [comment]
+1 Essential!

**Comment by rtrind** [comment]
+1, coming from a reddit topic where this was linked as the fix for the isolation of aliases being used as login alternatives from the main mail.

**Comment by JowanWorth** [comment]
+1 actually this does matter, the alias is only practical  at the moment is the account-level display name is removed entirely or set to something bland.

**Comment by exander77** [comment]
What is the status? This is is really need for almost everybody to use their Tutanota account. It has been open for a year and a half.

**Comment by JimmyCushnie** [comment]
I'm honestly baffled that this isn't in yet. Tutanota must not understand how important this feature is for business use.

**Comment by Joeyzo** [comment]
+1

**Comment by jones137** [comment]
I am also really in need of this feature, since I need my aliases to be confidential (not containing my name and surname). Changing back and forth is ridiculous.

**Comment by 4jNsY6fCVqZv** [comment]
What does it take from a technical/developer perspective to implement this feature?

**Comment by 4jNsY6fCVqZv** [comment]
What does it take from a technical/developer perspective to implement this feature and thus solve the issue?

**Comment by JimmyCushnie** [comment]
Basically nothing for the backend. The more difficult part is making the user interface for it.

**Comment by charlag** [comment]
We need to make a database change, that's the main part. If there would be a UI it would be nice but that's not the biggest problem.

**Comment by JimmyCushnie** [comment]
Why is it difficult to do something like this (pseudocode):
```
void SendEmail(EmailAddress senderAddress)
{
    var senderName = database.senderNames[senderAddress]
    Send(senderAddress, senderName)
}
```
I have almost no experience working with Tutanota-like programs, but I don't see why it would be hard at all to add alias names to the database.

**Comment by charlag** [comment]
Short answer is that we need to move name from `GroupInfo` (if I remember correctly). There's currently no place to put them so we need to create this mapping. This may affect some other things like user management and creation so it ripples.
In theory it's not hard but it's something that needs time investment.

**Comment by JimmyCushnie** [comment]
Fair enough.

I hope you get to this soon. For months I have been switching my "Sender Name" 2-3 times per day, which is quite annoying, not to mention error-prone.

**Comment by charlag** [comment]
I get it (and can relate to this), thanks for the patience!

**Comment by CyberAndre** [comment]
Me too is desperate for this feature. I can't get it together. Not having this feature, yet thinking of privacy.

**Comment by goldsolitude** [comment]
Another 1+ for this feature. Looking forward to when it is introduced. Until then I wont be able to use my real name on this service as I always go by my internet pseudonym wherever I sign up.

**Comment by finzzz** [comment]
2 years and no progress, and they are making useless features like calendar. This and crypto payment are far more important than that.

**Comment by devJ0n** [comment]
+1

**Comment by masvil** [comment]
I've spent 1 hour looking for a way to change sender name according to alias. I've been thinking "it MUST be somewhere". Then I gave up, googled it and... here I am with my + 1.

**Comment by ryan-boder** [comment]
This is a big problem. Please prioritize this.

**Comment by JimmyCushnie** [comment]
Seconded. It boggles my mind that this hasn't been added yet.

**Comment by akg83** [comment]
1+

This is an important feature, in order to make the email aliases feature work for real life usages.

**Comment by red2tech** [comment]
Actually there is a way to personalize the displayed name : and that is to not set up any account sender name (delete the current sender name or leave it blanck)

So, the system is forced to use whatever username you set up either @tutanota.com or @my_custom_domain.com

It's not perfect, as it's in all lowercase letters, but still it's better than a privacy breach

**Comment by masvil** [comment]
> It's not perfect, as it's in all lowercase letters, but still it's better than a privacy breach

Definitely! Good tip, thank you. That's a useful workaround waiting for a proper aliases' development.

I still wonder why this "feature" has still to be begged after 6 years instead of being considered as basic for any standard email service.

**Comment by HitchRait** [comment]
I know that this feature is still planned as many years before, my request is to **re-prioritize it**... 
I just learned I've exposed sender name to those who should not know it. Pain.

**Comment by vaf-hub** [comment]
It would probably be nice to also allow the configuration of different signatures depending on the chosen sender alias. I think, this can be an issue of its own, but it would be good to keep this in mind when making the required model changes.

**Comment by schipplock** [comment]
Och nö, jetzt wollte ich endlich mal zu Tutanota umziehen und finde das hier. Ich hatte angenommen so etwas sei eine Standardfunktion. Ich bin enttäuscht :(.

**Comment by schipplock** [comment]
Ich habe mich trotzdem für Tutanota entschieden. Die Vorteile überwiegen. Dass Kommentare ausgeblendet werden...nun ja...
